src/store-core/src/main/java/org/eclipse/lyo/store/internals/SparqlStoreImpl.java

```
@@ -229,6 +251,46 @@ public Model getJenaModelForSubject(final URI namedGraphUri, final URI subject)
         }
     }
 
+    @Override
+    public Model getResources(final URI namedGraph, final String prefixes, final String where, final int limit, final int offset) throws URISyntaxException {
+
+    	if (namedGraph != null) {
+        	//Make sure the designated namedGraph exists, if it is specified.
+    		//Otherwise, the search occurs across all namedgraphs.
+        	if (!namedGraphExists(namedGraph)) {
+                throw new IllegalArgumentException("Named graph" + String.valueOf(namedGraph) + " was missing from the triplestore");
+            }
+    	}
+
+    	SelectBuilder sparqlWhereQuery = constructSparqlWhere (prefixes, where, limit, offset);
+    	DescribeBuilder describeBuilder = new DescribeBuilder();
+		if (namedGraph != null) {
+			describeBuilder.addVar("s")
+			.addGraph(new ResourceImpl(String.valueOf(namedGraph)), sparqlWhereQuery);
+		}
+		else {
+			describeBuilder.addVar("s")
+			.addGraph("?g", sparqlWhereQuery);
+		}
+
+		Query describeQuery = describeBuilder.build() ;
+		String describeQueryString = describeQuery.toString();
+		final QueryExecution queryExecution = queryExecutor.prepareSparqlQuery(describeQueryString);
+        
+        Model execDescribe;
+        try {
+            execDescribe = queryExecution.execDescribe();
+		} catch (RiotException e) {
+			//a request that returns an empty set seems to cause an exception when using Marklogic.
+			if ((e.getCause() == null) && (e.getMessage().equals("[line: 2, col: 2 ] Out of place: [DOT]"))) {
```

yeah this is a crazy part but nothing we can do about it. ML returns `\n.\n` response, which puzzles Jena parser.

---

@jadelkhoury I think a better comment here could help. Let's say that ML returns `\n.\n` and that's why we are swallowing that exception.